tests/test_kernel_language.py

```diff
@@ -443,20 +454,32 @@ def test_EOSseawaterproperties_kernels(pset_mode, mode):
     class PoTempParticle(ptype[mode]):
         potemp = Variable('potemp', dtype=np.float32)
         pressure = Variable('pressure', dtype=np.float32, initial=10000)
-    pset = pset_type[pset_mode]['pset'](fieldset, pclass=PoTempParticle, lon=5, lat=5, depth=1000)
+    if pset_mode != 'nodes':
+        pset = pset_type[pset_mode]['pset'](fieldset, pclass=PoTempParticle, lon=5, lat=5, depth=1000)
+    else:
+        pset = pset_type[pset_mode]['pset'](fieldset, pclass=PoTempParticle, lon=5, lat=5, depth=1000,
+                                            idgen=idgen, c_lib_register=c_lib_register)
```

Comment:
(a) I wrote the that test before I introduced the `kwargs` default option.
(b) I also want(ed) to demo the different ways to use the data structure.

Reply:
ah, now, after a closer look on that test: that test reuses the `pset` variable and in order to avoid recreating the `LibraryInterfaceC` over and over again (which takes quite some time and may yields deletion failures on the particle set), the test uses 1 global instance of the C-interface and the ID generator - no defaults.

On your specific question: It is actually something I needed to do for the benchmarks, but that is part of a different PR (and I'm not gonna mix them now). It takes some redesign effort because, if `idgen` and `c_lib_register` are part of the `kwargs` for AOS and SOA, then they need to be forcefully popped from the `kwargs` argument list at the start of their `__init__` functions - otherwise the `__init__` function attempts to interpret them as per-particle attributes, which they are not, resulting in a lot of problems. For the nodes, that is no problem because there, the two variables are named, expected variables (i.e. they are not actually part of the `kwargs` variable of `__init__`), so there is no confusion what those variables shall be.